packages/medusa/src/repositories/product.ts

```diff
@@ -213,11 +214,13 @@ export const ProductRepository = dataSource.getRepository(Product).extend({
       customJoinBuilders: [
         (queryBuilder, alias, topLevel) => {
           if (topLevel === "variants") {
-            queryBuilder.leftJoinAndSelect(
-              `${alias}.${topLevel}`,
-              topLevel,
-              `${topLevel}.deleted_at IS NULL`
-            )
+            const joinMethod = select.filter(
+              (key) => !!key.match(/^variants\.\w+$/i)
+            ).length
+              ? "leftJoin"
+              : "leftJoinAndSelect"
+
+            queryBuilder[joinMethod](`${alias}.${topLevel}`, topLevel)
```

Comment:
comment: Just pointing out that you've removed the `deleted_at is NULL` clause. Don't know if this is intentional.

Reply:
Yes, because it is automatically added by typeorm since there is a deleted_at column decorated. (`DeleteDateColumn`) Therefore it was the reason the constraint appeared twice for each query